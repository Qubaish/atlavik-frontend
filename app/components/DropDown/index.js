import React from 'react';
import PropTypes from 'prop-types';

function DropDown({val, onChange}) {
    return (
        <div className="row">
        <label>City</label>
          <select
            className="browser-default"
            onChange={onChange}
            value={val}
            name="city"
          >
            <option value="" disabled>Choose your option</option>
            <option value="dubai">Dubai</option>
            <option value="cairo">Cairo</option>
            <option value="london">London</option>
          </select>
        </div>
    )
}

DropDown.propTypes = {
    onChange: PropTypes.func,
    val: PropTypes.string,
};

export default DropDown;
