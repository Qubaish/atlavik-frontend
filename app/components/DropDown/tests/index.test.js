import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import DropDown from '../index';

describe('<DropDown />', () => {
    it('should render a DropDown', () => {
        const container = shallow(
            <DropDown val="Dubai" onChange={jest.fn()} />
        );
        expect(shallowToJson(container)).toMatchSnapshot();
    });

    it('should called onChange', () => {
        const submitSpy = jest.fn();
        const container = shallow(
            <DropDown val="Dubai" onChange={submitSpy} />
        );
        container.find('select').props().onChange();
        expect(submitSpy).toHaveBeenCalled();
    });
});
