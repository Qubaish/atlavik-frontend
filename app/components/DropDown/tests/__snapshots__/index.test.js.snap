// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropDown /> should render a DropDown 1`] = `
<div
  className="row"
>
  <label>
    City
  </label>
  <select
    className="browser-default"
    name="city"
    onChange={[MockFunction]}
    value="Dubai"
  >
    <option
      disabled={true}
      value=""
    >
      Choose your option
    </option>
    <option
      value="dubai"
    >
      Dubai
    </option>
    <option
      value="cairo"
    >
      Cairo
    </option>
    <option
      value="london"
    >
      London
    </option>
  </select>
</div>
`;
